src/cc/frontends/clang/kbuild_helper.cc

```diff
@@ -140,15 +140,22 @@ int KBuildHelper::get_flags(const char *uname_machine, vector<string> *cflags) {
   return 0;
 }
 
-static inline int file_exists(const char *f)
+static inline int file_exists_and_ownedby(const char *f, uid_t uid)
 {
   struct stat buffer;
-  return (stat(f, &buffer) == 0);
+  int ret;
+  if ((ret = stat(f, &buffer)) == 0) {
+    if (buffer.st_uid != uid) {
+      std::cout << "ERROR: header file ownership unexpected: " << std::string(f) << "\n";
```

Comment:
Nit: is it useful to also print out the expected `uid` and the one in the file: `st_uid`?

Reply:
I think this is close enough: says this filename has this type of error. End-user should likely see this and run `ls -l`. There's related codepaths that can error with no such hint, which is pretty hard to debug, which is why I put a big clue in. :-)